### PR TITLE
Allow CSV headers to contain special characters.

### PR DESCRIPTION
**Pull request by illegalprime**
sorry for the white space changes, I didn't notice them

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -460,7 +462,7 @@ apiRouter.route("/data/tag/:tag").delete(authenticateWithReject, async (request,
 
 apiRouter.route("/search").get(authenticateWithReject, async (request, response) => {
 	let query: string = request.query.q || "";
-	let queryRegExp = new RegExp(query, "i");
+	let queryRegExp = new RegExp(reEscape(query), "i");
```
Comment (illegalprime) [inline server/app.ts]: I don't actually know if this should be escaped or not.

Comment (petschekr) [inline server/app.ts]: I think it should because query is a string that can contain characters with special meaning for the regex parser

Comment (illegalprime) [inline server/app.ts]: oh OK I'll remove this

Comment (petschekr) [inline server/app.ts]: Wait no don't remove it unless we want that to be a feature (regex searching)

Comment (illegalprime) [inline server/app.ts]: already did :(

Comment (petschekr) [inline server/app.ts]: Welp that bug is now a feature, cool :+1:

**Pull request merged**